### PR TITLE
Add user select homepage plugin

### PR DESCRIPTION
**Pull request by charrondev**
This is a simple plugin to allow users to check set their preferred home page separately from the site's default settings.

The plugin is made up of the following:

-   A custom preferences page on the profile controller with a simple form.
-   Logic for saving that value with the `UserMetaModel`.
-   An event handler for dynamically setting the default controller with `Gdn_Router` when the dispatcher is started.
-   Event handlers adding navigation to the user preference pages.

### CODE REVIEW

File: plugins/userhomepage/UserHomePagePlugin.php [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+        $sender->editMode(true);
+        $sender->title(t('Home Page Settings'));
+        // Form initialization
+        $homeUrl =  url('/', true);
+        $formTemplate = [
+            'homepage' => [
+                'LabelCode' => 'Home Page',
+                "Description" => sprintf(
+                    t("Choose the page you would like to see when you visit visit: %s."),
```
Comment (igraziatto) [inline plugins/userhomepage/UserHomePagePlugin.php]:
```suggestion
                    t("Choose the page you would like to see when you visit: %s."),
```

File: plugins/userhomepage/UserHomePagePlugin.php [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+    /**
+     * Determine if the sessioned user can edit the home page settings of a particular user.
+     *
+     * @param \ProfileController $profileController
+     * @return bool
+     */
+    private function canEditUsersPreference(\ProfileController $profileController): bool {
+        $userID = $profileController->User->UserID;
```
Comment (igraziatto) [inline plugins/userhomepage/UserHomePagePlugin.php]: `$profileController->User` can be either an object or array here, this is throwing an error:

Screen Shot 2020-01-17 at 2 14 52 PM

File: plugins/userhomepage/views/homepage.twig [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<style>
+    {# Ewwww #}
```
Comment (igraziatto) [inline plugins/userhomepage/views/homepage.twig]: Addons is a public repo, do you really wanna have those type of comments?

File: plugins/userhomepage/UserHomePagePlugin.php [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        $formTemplate = [
+            'homepage' => [
+                'LabelCode' => 'Home Page',
+                "Description" => sprintf(
+                    t("Choose the page you would like to see when you visit visit: %s."),
+                    "<strong><a href='$homeUrl'>$homeUrl</a></strong>"
+                ),
+                'Control' => 'RadioList',
+                'Items' => [
+                    '/categories' => '/categories',
```
Comment (igraziatto) [inline plugins/userhomepage/UserHomePagePlugin.php]:
```suggestion
                    '/categories' => 'categories',
```

File: plugins/userhomepage/UserHomePagePlugin.php [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+                    '/discussions' => '/discussions',
```
Comment (igraziatto) [inline plugins/userhomepage/UserHomePagePlugin.php]:
```suggestion
                    '/discussions' => 'discussions',
```

File: plugins/userhomepage/views/homepage.twig (hunk 3 above)
Comment (charrondev) [inline plugins/userhomepage/views/homepage.twig]:
```suggestion
```

**Pull request closed without merging**